MixItUp.Base/ViewModel/User/UserViewModel.cs

```diff
@@ -470,9 +398,66 @@ public int CompareTo(UserViewModel other)
 
         private void SetMixerRoles(string[] userRoles)
         {
-            if (userRoles != null && userRoles.Count() > 0)
+            lock (this.mixerRoles)
             {
-                this.chatRoles = new List<string>(userRoles);
+                this.mixerRoles.Clear();
+                this.mixerRoles.Add(MixerRoleEnum.User);
+
+                if (userRoles.Any(r => r.Equals("Owner"))) { this.mixerRoles.Add(MixerRoleEnum.Streamer); }
```

Comment:
Crud, I need a null check here.

Reply:
And throw the Count() check in there as well, I've seen some slightly wonky behavior from Mixer where we get an empty role list back, so I'd like to just skip it all if it happens. To give some buffering to it, I'd put the null & Count() check AFTER line 404. This will ensure every person at least gets the User role added for them if something craps out.

Reply:
Yup, got it. :)